config/v1/types_infrastructure.go

```diff
@@ -123,6 +123,17 @@ const (
 	ExternalTopologyMode TopologyMode = "External"
 )
 
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:validation:Enum:="Internal";"External"
+type PlatformLoadBalancerType string
+
+const (
+	// LoadBalancerTypeExternal is a load balancer that is managed outside of the cluster.
+	LoadBalancerTypeExternal PlatformLoadBalancerType = "CustomerManaged"
```

Comment:
I think we were supposed to talk about this in the meeting earlier, but if we don't like the internal and external terminology should we just use CustomerManaged and OpenShiftManaged everywhere?

Reply:
I'll let David or Joel to comment on that.
Please let me know exactly what's expected though so I can apply the necessary changes.

Reply:
I like `External` as a description of the intended management of external LB. It's analogous to e.g. external CCM: it's managed, just not by us.

I agree `Internal` could be more descriptive.

Whatever we choose, please could we not use `Customer` here? It's weird, as well as being wrong in many circumstances.

Reply:
We've bifurcated this discussion across 2 PRs now. [Ref](https://github.com/openshift/enhancements/pull/1322#discussion_r1090524302) to @patrickdillon's original comment

@patrickdillon The `internal` name is normally referring to something which ends up as `api-int` right? What is the name we use to refer to the other LB, with the `api` DNS record? Is it `public`?

Should we use the name of the software that runs in the cluster for the internal version? `ClusterKeepalived`? `ClusterHAproxy`? What other thoughts do people have re the naming?

Reply:
> Should we use the name of the software that runs in the cluster for the internal version? ClusterKeepalived? ClusterHAproxy?

@cybertron might have an opinion on this one.
My take on this is one field should be enough for our need. I don't foresee the separation of Keepalived & HAproxy, but I might be wrong?

As for naming, my understanding is that `UserManaged` and `OpenShiftManaged` were ok for everyone. However I'm not sure about `LoadBalancerTypeExternal` and `LoadBalancerTypeInternal` which is where I'm looking for feedback.

Reply:
> I think we were supposed to talk about this in the meeting earlier, but if we don't like the internal and external terminology should we just use CustomerManaged and OpenShiftManaged everywhere?

@cybertron was your comment for `LoadBalancerTypeExternal` & `LoadBalancerTypeInternal` ? or the field value only? or both

Reply:
> @patrickdillon The `internal` name is normally referring to something which ends up as `api-int` right? What is the name we use to refer to the other LB, with the `api` DNS record? Is it `public`?

"Internal" is used pretty consistently; the other LB naming is a bit less consistent, we kind of use "external", "public", "API", or any combination thereof. It also depends on the platform: Azure & GCP use the term "external" to varying degrees in their docs, while I don't see it in AWS docs.